src/web/app/components/question-types/question-constraint/question-constraint.component.ts

```diff
@@ -0,0 +1,12 @@
+import { EventEmitter, Output } from '@angular/core';
+import { Observable } from 'rxjs';
+
+/**
+ * Base class for constraint components.
+ */
+export abstract class QuestionConstraintComponent {
```

Comment:
with the help of generic class
```suggestion
export abstract class QuestionConstraintComponent<Q extends FeedbackQuestionDetails> {
```

And therefore the `isValid` check can be here (The child component will only need care about whether the `isValid` or not. They don't need to fire event!).